src/fromager/commands/build.py

```diff
@@ -482,16 +478,21 @@ def _is_wheel_built(
             raise ValueError(
                 f"{dist_name}: changelog for version {resolved_version} is inconsistent. Found build tag {existing_build_tag} but expected {build_tag}"
             )
-        is_built = existing_build_tag == build_tag
-        if is_built and wkctx.wheel_server_url not in url:
+        if existing_build_tag != build_tag:
+            logger.info(
+                f"existing build tag {existing_build_tag} does not match expected build tag {build_tag}"
+            )
+            return None
+
+        if wkctx.wheel_server_url not in url:
```

Comment:
I found this check suspicious when I looked through the looks. How about `if not url.startswith(wkctx.wheel_server_url)` + a bit of extra debugging logging in the true and false branch?

Reply:
Good point

Reply:
Updated